src/libstd/memchr.rs

```diff
@@ -209,7 +209,7 @@ mod fallback {
         let end_align = (ptr as usize + len) & (usize_bytes - 1);
         let mut offset;
         if end_align > 0 {
-            offset = len - cmp::min(usize_bytes - end_align, len);
+            offset = len - cmp::min(end_align, len);
```

Comment:
This seems correct to me, but I think the code would be somewhat clearer as explicit if:
```
offset = if end_align > len { 0 } else { len - end_align };
```